invokeai/backend/prompting/conditioning.py

```diff
@@ -25,58 +26,52 @@
-    tokenizer = model.tokenizer
-    compel = Compel(
-        tokenizer=tokenizer,
-        text_encoder=model.text_encoder,
-        textual_inversion_manager=model.textual_inversion_manager,
-        dtype_for_device_getter=torch_dtype,
-        truncate_long_prompts=False
-    )
+    compel = Compel(tokenizer=model.tokenizer,
+                    text_encoder=model.text_encoder,
+                    textual_inversion_manager=model.textual_inversion_manager,
+                    dtype_for_device_getter=torch_dtype,
+                    truncate_long_prompts=False,
+                    )
 
     # get rid of any newline characters
     prompt_string = prompt_string.replace("\n", " ")
-    (
-        positive_prompt_string,
-        negative_prompt_string,
-    ) = split_prompt_to_positive_and_negative(prompt_string)
-    legacy_blend = try_parse_legacy_blend(
-        positive_prompt_string, skip_normalize_legacy_blend
-    )
-    positive_prompt: Union[FlattenedPrompt, Blend]
+    positive_prompt_string, negative_prompt_string = split_prompt_to_positive_and_negative(prompt_string)
+
+    legacy_blend = try_parse_legacy_blend(positive_prompt_string, skip_normalize_legacy_blend)
+    positive_conjunction: Conjunction
     if legacy_blend is not None:
-        positive_prompt = legacy_blend
+        positive_conjunction = legacy_blend
     else:
-        positive_prompt = Compel.parse_prompt_string(positive_prompt_string)
-    negative_prompt: Union[FlattenedPrompt, Blend] = Compel.parse_prompt_string(
-        negative_prompt_string
-    )
+        positive_conjunction = Compel.parse_prompt_string(positive_prompt_string)
+    positive_prompt = positive_conjunction.prompts[0]
 
-    if log_tokens or getattr(Globals, "log_tokenization", False):
-        log_tokenization(positive_prompt, negative_prompt, tokenizer=tokenizer)
-
-    c, options = compel.build_conditioning_tensor_for_prompt_object(positive_prompt)
-    uc, _ = compel.build_conditioning_tensor_for_prompt_object(negative_prompt)
-    [c, uc] = compel.pad_conditioning_tensors_to_same_length([c, uc])
-
-    tokens_count = get_max_token_count(tokenizer, positive_prompt)
+    negative_conjunction = Compel.parse_prompt_string(negative_prompt_string)
+    negative_prompt: FlattenedPrompt | Blend = negative_conjunction.prompts[0]
 
-    ec = InvokeAIDiffuserComponent.ExtraConditioningInfo(
-        tokens_count_including_eos_bos=tokens_count,
-        cross_attention_control_args=options.get("cross_attention_control", None),
-    )
+    tokens_count = get_max_token_count(model.tokenizer, positive_prompt)
+    if log_tokens or getattr(Globals, "log_tokenization", False):
+        log_tokenization(positive_prompt, negative_prompt, tokenizer=model.tokenizer)
+
+    with InvokeAIDiffuserComponent.custom_attention_context(model.unet,
+                                                            extra_conditioning_info=None,
+                                                            step_count=-1):
```

Comment:
What's this custom attention context for?

and, uh, how is this working? This is calling that as if it's a classmethod, but it's not.

Reply:
@keturn This was the result of an incomplete cut-and-paste from the changes that @damian0815 made in the v2.3 branch to support the latest Compel. I think it's fixed in my latest commit to this PR. However, I defer to Damian for what exactly this context is doing .

Reply:
In original code here pass extra conditioning info with loras and this just applies loras to get right results from text_encoder.
Now we not support loras and it can be removed

Reply:
i also do not understand this. it seems like a messed up merge.

Reply:
@keturn i had a proper look and jogged my memory, it has been de-state-ified to a classmethod on the v2.3 branch but is still an instance method here

Reply:
I commented it out and left a TODO.